# Covid19_ONT_Artic

**Covid19_ONT_Artic** is a set of scripts for running the [ARTIC pipeline](https://github.com/artic-network/fieldbioinformatics) from fast5 ONT data to filtered SARS-CoV-2 variants and coverage plots. Example data for sample SP1, the first case in Sao Paulo (Brasil), can be downloaded from [here](https://cadde.s3.climb.ac.uk/SP1-raw.tgz).

<p align="center">
  <img src="Figures/Example_Coverage_Plot.png" alt="drawing" width="600" title="Coverage plot">
</p>

## Getting started

**Prerequisites**

* Miniconda3.
Tested with conda 4.9.2.
```which conda``` should return the path to the executable.
If you don't have Miniconda3 installed, you could download and install it with:
```
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
chmod 755 Miniconda3-latest-Linux-x86_64.sh
./Miniconda3-latest-Linux-x86_64.sh
```

Then, after completing _Covid19_ONT_Artic_ installation, set the _MINICONDA_DIR_ variable in **config_Covid19_ONT_Artic.R** to the full path to miniconda3 directory.

* Guppy, the software for basecalling and demultiplexing provided by ONT. Tested with Guppy v4.2.
If you don't have [Guppy](https://community.nanoporetech.com/downloads) installed, choose an appropriate version and install it.
For example, you could download and unpack the archive with:
```
wget /path/to/ont-guppy-cpu_version_of_interest.tar.gz
tar -xf ont-guppy-cpu_version_of_interest.tar.gz
```
A directory _ont-guppy-cpu_ should have been created in your current directory.
Then, after completing _Covid19_ONT_Artic_ installation, set the _BASECALLER_DIR_ variable in **config_Covid19_ONT_Artic.R** to the full path to _ont-guppy-cpu/bin_ directory.

**Installation**

```
git clone https://github.com/MaestSi/Covid19_ONT_Artic.git
cd Covid19_ONT_Artic
chmod 755 *
./install.sh
```

A conda environment named _Covid19_ONT_Artic_env_ is created, where seqtk, NanoFilt, Bedtools and R with packages biostrings, ggplot2 and scales are installed. Moreover, a conda environment named _pycoQC_env_ is created, where pycoQC is installed.
Then, you can open the **config_Covid19_ONT_Artic.R** file with a text editor and set the variables _PIPELINE_DIR_ and _MINICONDA_DIR_ to the value suggested by the installation step.

## Usage

**Launch_Covid19_ONT_Artic.sh**

Usage: Launch_Covid19_ONT_Artic.sh \<fast5_dir\>

Input:
* \<fast5_dir\>: directory containing raw fast5 files

Outputs (saved in \<fast5_dir\>\_analysis/analysis):
* logfile.txt: log file describing the progress of the workflow
* \<sample_name\> directory containing:
  * Files generated by ARTIC pipeline
  * \<sample_name\>\_linear.dat: file containing average coverage across regions defined in BED_COV file
  * \<sample_name\>.dat: file containing average coverage in log scale across regions defined in BED_COV file
  * \<sample_name\>\_PrimerCov.png: plot representing average coverage in log scale across regions defined in BED_COV file
  * \<sample_name\>.pass_intersected_BED_VAR.vcf: file obtained intersecting \<sample_name\>.pass.vcf.gz file and variants of interest defined in BED_VAR file
  * \<sample_name\>\_NOT\_GENOTYPED\_VAR.bed: file containing variants of interest defined in BED_VAR file which could not be genotyped

Outputs (saved in \<fast5_dir\>\_analysis/qc):
* Read length distributions and pycoQC report

Outputs (saved in \<fast5_dir\>\_analysis/basecalling):
* Temporary files for basecalling

Outputs (saved in \<fast5_dir\>\_analysis/preprocessing):
* Temporary files for demultiplexing, filtering based on read length and adapters trimming

## Auxiliary scripts

In the following, auxiliary scripts run by **Launch_Covid19_ONT_Artic.sh** are listed. These scripts should not be called directly.

**Covid19_ONT_Artic.R**

Note: script run by _Launch_Covid19_ONT_Artic.sh_.

**config_Covid19_ONT_Artic.R**

Note: configuration script, must be modified before running _Launch_Covid19_ONT_Artic.sh_.


**subsample_fast5.sh**

Note: script run by _Covid19_ONT_Artic.R_ if _do_subsampling_flag_ variable is set to 1 in _config_Covid19_ONT_Artic.R_.
